### PR TITLE
Add ability to edit participant name and participant types in Manage Participants Table

### PR DESCRIPTION
**Pull request by ashleysmithTTD**
- Update participant name and/or participant types and salesforce agreement number
- Participant must have at least one participant type checked and at least one API role checked for update to be successful, error will display if not
- If salesforce agreement number already exists and participant edits it, it validates that it should have 8 digits
- Changed where participant types data is being pulled (was Admin, now portal db)
- Updated component in Storybook
- Added Readonly to props where SonarLint suggested
- Made title of Update Dialog a little clearer

### CODE REVIEW

File: src/api/routers/participants/participantsRouter.ts [hunk 1]
```diff
@@ -127,23 +125,7 @@ export function createParticipantsRouter() {
...
-        const currentSite = p?.siteId === undefined ? undefined : siteMap.get(p.siteId);
-        return {
-          ...p,
-          types: mapClientTypeToParticipantType(
-            currentSite?.clientTypes || [],
-            allParticipantTypes
-          ),
-        };
-      })
-      .sort((a, b) => a.name.localeCompare(b.name));
```
Comment (ashleysmithTTD) [inline src/api/routers/participants/participantsRouter.ts:146]: This function was pulling data from AdminDB, but we want to portal DB  info instead.  This made the code much simpler and also removed several imports.

File: src/web/components/ParticipantManagement/UpdateParticipantDialog.tsx [hunk 2]
```diff
@@ -3,49 +3,79 @@ import { SubmitHandler } from 'react-hook-form';
...
 }: UpdateParticipantDialogProps) {
   const [open, setOpen] = useState(false);
+  const [typesAndRolesError, setTypesAndRolesError] = useState(false);
 
   const onSubmit: SubmitHandler<UpdateParticipantForm> = async (formData) => {
-    await onUpdateParticipant(formData, participant);
-    setOpen(false);
+    if (formData.participantTypes.length === 0 || formData.apiRoles.length === 0) {
+      setTypesAndRolesError(true);
+    } else {
```
Comment (ashleysmithTTD) [inline src/web/components/ParticipantManagement/UpdateParticipantDialog.tsx:35]: Per Kimberly, we should not be able to update a participant that removes all participant types or remove all API roles.  Participants must have at least one of each.

**Comment by ashleysmithTTD** [comment]
https://github.com/IABTechLab/uid2-self-serve-portal/assets/157655209/219cf2b1-d966-4e05-9ea0-fc1867a192ba

File: src/web/components/ParticipantManagement/UpdateParticipantDialog.tsx [hunk 3]
```diff
@@ -29,23 +34,45 @@ function UpdateParticipantDialog({
...
       <Form<UpdateParticipantForm>
         onSubmit={onSubmit}
         submitButtonText='Save Participant'
         defaultValues={originalFormValues}
       >
+        <TextInput
+          inputName='participantName'
+          label='Participant Name'
+          className='text-input'
+          required
```
Comment (ssundahlTTD) [inline src/web/components/ParticipantManagement/UpdateParticipantDialog.tsx:65]: let's do rules={{ required: '...' }} like the other components on the page for consistency.

**Review by ssundahlTTD — APPROVED** [review]

**Pull request merged**